### PR TITLE
[FLINK-11645] Remove legacy JobManager 

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

This commit removes the legacy `JobManager` component from Flink.

This PR is based on #7725 and #7694.

## Brief change log

- Remove `TestingUtils#createJobManager`
- Remove `TestingJobManagerMessages`
- Remove `TestingJobManager`
- Remove `JobManager`

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskManagerRegistrationTest.java [hunk 1]
```diff
@@ -1,694 +0,0 @@
-/*
```
Comment (GJL) [inline flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskManagerRegistrationTest.java]: Commit message should be fixed before merging.

File: flink-runtime/src/main/java/org/apache/flink/runtime/messages/checkpoint/TriggerCheckpoint.java [hunk 2]
```diff
@@ -84,7 +84,11 @@ else if (o instanceof TriggerCheckpoint) {
 
 	@Override
 	public String toString() {
-		return String.format("Trigger Checkpoint %d@%d for (%s/%s)", 
-				getCheckpointId(), getTimestamp(), getJob(), getTaskExecutionId());
+		return String.format(
```
Comment (GJL) [inline flink-runtime/src/main/java/org/apache/flink/runtime/messages/checkpoint/TriggerCheckpoint.java]: nit: checkstyle hotfix not related to _removing legacy JobManager_.

Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/messages/checkpoint/TriggerCheckpoint.java]: True, I'll revert it.

Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/messages/checkpoint/TriggerCheckpoint.java]: True, I'll make it a separate commit.

**Pull request closed without merging**